src/utils/utilities.ts

```diff
@@ -240,3 +244,189 @@ export const calculateTimeLeft = (createdAt: string) => {
+
+    const signersList = [];
+    let signerMatched = false;
+
+    psbt.data.inputs.forEach((input) => {
+      if (input.bip32Derivation) {
+        // Loop through all derivations (in case there are multiple keys)
+        input.bip32Derivation.forEach((derivation) => {
+          const data = {
+            derivationPath: shortDerivationPath(derivation.path),
+            masterFingerprint: derivation.masterFingerprint.toString('hex'),
+            xpriv: '',
+            xpub: derivation.pubkey.toString('hex'),
+          };
+          signersList.push(data);
+          if (data.masterFingerprint.toLowerCase() === signer.masterFingerprint.toLowerCase()) {
+            signerMatched = true;
+          }
+
+          // const val = WalletUtilities.getFingerprintFromExtendedKey(
+          //   derivation.pubkey.toString('hex'),
+          //   network
+          // );
+        });
+      }
+    });
+
+    // Check if signer exists in the psbt
+    const currentSignerFromPSBT = signersList.find(
+      (psbtSigner) => psbtSigner.masterFingerprint === signer.masterFingerprint.toLowerCase()
+    );
+
+    // Extract input addresses
+    const senderAddresses = psbt.txInputs.map((input) => input.hash.toString('hex'));
+
+    // Extract outputs (receiver information)
+    const outputs = psbt.txOutputs.map((output) => {
+      return {
+        address: bitcoin.address.fromOutputScript(
+          output.script,
+          isTestnet() ? bitcoin.networks.testnet : bitcoin.networks.bitcoin
+        ), // Receiver address
+        amount: output.value, // Amount in satoshis
+      };
+    });
+
+    // Calculate the total input and output amounts
+    let totalInput = 0;
+    let totalOutput = 0;
+    let totalAmount = 0;
+
+    psbt.data.inputs.forEach((input, index) => {
+      if (input.witnessUtxo) {
+        totalInput += input.witnessUtxo.value;
+      } else if (input.nonWitnessUtxo) {
+        const tx = bitcoin.Transaction.fromBuffer(input.nonWitnessUtxo);
+        const voutIndex = psbt.txInputs[index].index;
+        totalInput += tx.outs[voutIndex].value;
+      }
+    });
+
+    outputs.forEach((output) => {
+      totalOutput += output.amount;
+      if (!changeAddress.includes(output.address)) {
+        totalAmount += output.amount;
+      }
+    });
+
+    const receiverAddresses = outputs.map((op) => op.address);
+
+    // Calculate transaction fees
+    const fees = totalInput - totalOutput;
+    const feeRate = (fees / vBytes).toFixed(2);
+    return {
+      senderAddresses: senderAddresses,
+      receiverAddresses: receiverAddresses,
+      fees,
+      sendAmount: totalAmount,
+      currentSignerFromPSBT,
+      signerMatched,
+      changeAddress,
+      txId,
+      vout,
+      feeRate,
+      vBytes,
+    };
+  } catch (error) {
+    console.log('🚀 ~ dataFromPSBT ~ error:', error);
+    throw 'Something went wrong';
+  }
+};
+
+export const estimateVByteFromPSBT = (base64Str: string) => {
```

Comment:
This is also not correct since the transaction will be bigger with signatures. What you should do here is estimate the final size of the transaction, not check the current one.

See https://jlopp.github.io/bitcoin-transaction-size-calculator/

However since this is somewhat complicated we could show just the total fee instead of sats/vbyte and work on that later

Reply:
@ben-kaufman  Exact transaction size cannot be calculated without finalising the psbt. Hence we are estimating the size, similar to what sparrow does.

Reply:
But this estimation is not taking into account signatures, so it's way too inaccurate. It's be good to have a proper estimation. I'm just suggesting that for simplicity, we could do it later.